pkg/cmd/cli/cmd/rollout/retry.go

```diff
@@ -0,0 +1,192 @@
+
+	Filenames []string
+	Out       io.Writer
+	Recursive bool
+
+	Clientset *kclientset.Clientset
+}
+
+var (
+	rolloutRetryLong = templates.LongDesc(`
+		If a rollout fails, you may opt to retry it (if the error was transient). Some rollouts may
+		never successfully complete - in which case you can use the rollout latest to force a redeployment.
+		If a deployment config has completed rolling out successfully at least once in the past, it would be
+		automatically rolled back in the event of a new failed rollout. Note that you would still need
+		to update the erroneous deployment config in order to have its template persisted across your
+		application.
+`)
+
+	rolloutRetryExample = templates.Examples(`
+	  # Retry the latest failed deployment based on 'frontend'
+	  # The deployer pod and any hook pods are deleted for the latest failed deployment
+	  %[1]s rollout retry dc/frontend
+`)
+)
+
+func NewCmdRolloutRetry(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	opts := &RetryOptions{}
+	cmd := &cobra.Command{
+		Use:     "retry (TYPE NAME | TYPE/NAME) [flags]",
+		Long:    rolloutRetryLong,
+		Example: fmt.Sprintf(rolloutRetryExample, fullName),
+		Short:   "Retry the latest failed rollout",
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(opts.Complete(f, cmd, out, args))
+			kcmdutil.CheckErr(opts.Run())
+		},
+	}
+	usage := "Filename, directory, or URL to a file identifying the resource to get from a server."
+	kubectl.AddJsonFilenameFlag(cmd, &opts.Filenames, usage)
+	kcmdutil.AddRecursiveFlag(cmd, &opts.Recursive)
+	return cmd
+}
+
+func (o *RetryOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, args []string) error {
+	if len(args) == 0 && len(o.Filenames) == 0 {
+		return kcmdutil.UsageError(cmd, cmd.Use)
+	}
+
+	o.Mapper, o.Typer = f.Object(false)
+	o.Encoder = f.JSONEncoder()
+	o.Out = out
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	_, _, o.Clientset, err = f.Clients()
+	if err != nil {
+		return err
+	}
+
+	r := resource.NewBuilder(o.Mapper, o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, o.Recursive, o.Filenames...).
+		ResourceTypeOrNameArgs(true, args...).
+		ContinueOnError().
+		Latest().
+		Flatten().
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+
+	o.Infos, err = r.Infos()
+	return err
+}
+
+func (o RetryOptions) Run() error {
+	allErrs := []error{}
+	mapping, err := o.Mapper.RESTMapping(kapi.Kind("ReplicationController"))
+	if err != nil {
+		return err
+	}
+	for _, info := range o.Infos {
+		config, ok := info.Object.(*deployapi.DeploymentConfig)
+		if !ok {
+			allErrs = append(allErrs, kcmdutil.AddSourceToErr("retrying", info.Source, fmt.Errorf("expected deployment configuration, got %T", info.Object)))
+			continue
+		}
+		if config.Spec.Paused {
```

Comment:
yeah, but we do have them in cancel as well.. agree that having them in validate and simply removing the infos for configs that does not pass them seems more clean. i will fix it as a follow up for both cancel and this.